### PR TITLE
[RLlib] Remove utilities/tests/classes not needed anymore.

### PR DESCRIPTION
**Pull request by sven1977**
Remove utilities/tests/classes not needed anymore.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kouroshHakha** [comment]
NOTE: premerge tests are failing.

File: rllib/examples/custom_keras_model.py
Comment (kouroshHakha) [inline rllib/examples/custom_keras_model.py:1]: why this one?

File: rllib/examples/custom_rnn_model.py
Comment (kouroshHakha) [inline rllib/examples/custom_rnn_model.py:1]: should we have an equivalent RLModule example for this?

File: rllib/examples/custom_torch_policy.py
Comment (kouroshHakha) [inline rllib/examples/custom_torch_policy.py:1]: We should have a customer EnvRunner equivalent for this?

File: rllib/examples/multi-agent-leela-chess-zero.py
Comment (kouroshHakha) [inline rllib/examples/multi-agent-leela-chess-zero.py:1]: Has this been moved to the contrib folder?

File: rllib/examples/preprocessing_disabled.py
Comment (kouroshHakha) [inline rllib/examples/preprocessing_disabled.py:1]: why?

File: rllib/examples/recommender_system_with_recsim_and_slateq.py
Comment (kouroshHakha) [inline rllib/examples/recommender_system_with_recsim_and_slateq.py:1]: why?

File: rllib/examples/remote_base_env_with_custom_api.py
Comment (kouroshHakha) [inline rllib/examples/remote_base_env_with_custom_api.py:1]: why?

File: rllib/examples/remote_envs_with_inference_done_on_main_node.py
Comment (kouroshHakha) [inline rllib/examples/remote_envs_with_inference_done_on_main_node.py:1]: why?

File: rllib/examples/vizdoom_with_attention_net.py
Comment (kouroshHakha) [inline rllib/examples/vizdoom_with_attention_net.py:1]: why?

**Review by kouroshHakha — COMMENTED** [review]
It's just not clear to me what the decision process behind removals are. I have asked why on some of those that I would probably not have deleted. Some explanation on the process is missing on the PR details.

File: rllib/examples/recommender_system_with_recsim_and_slateq.py
Comment (sven1977) [reply rllib/examples/recommender_system_with_recsim_and_slateq.py:1]: SlateQ is now in rllib_contrib.

File: rllib/examples/remote_base_env_with_custom_api.py
Comment (sven1977) [reply rllib/examples/remote_base_env_with_custom_api.py:1]: undone

File: rllib/examples/remote_envs_with_inference_done_on_main_node.py
Comment (sven1977) [reply rllib/examples/remote_envs_with_inference_done_on_main_node.py:1]: undone

File: rllib/examples/custom_torch_policy.py
Comment (sven1977) [reply rllib/examples/custom_torch_policy.py:1]: More like a custom Learner? The "custom" bit here is the loss function. Undone.

File: rllib/examples/custom_rnn_model.py
Comment (sven1977) [reply rllib/examples/custom_rnn_model.py:1]: Yes :) Undone

File: rllib/examples/custom_keras_model.py
Comment (sven1977) [reply rllib/examples/custom_keras_model.py:1]: Undone.

File: rllib/examples/vizdoom_with_attention_net.py
Comment (sven1977) [reply rllib/examples/vizdoom_with_attention_net.py:1]: undone

File: rllib/examples/multi-agent-leela-chess-zero.py
Comment (sven1977) [reply rllib/examples/multi-agent-leela-chess-zero.py:1]: Good catch. This is actually a "tuned_example". Created it in `rllib_contrib`. It wasn't run as a test in the old CI, but I activated it now. Let's see whether it learns a little bit (0.6 win rate vs random should be doable :) ).

**Comment by sven1977** [comment]
Thanks for your review @kouroshHakha !
I admit some of the example removals are debateable. The main thought was that some of the more advanced features that we don't really want users to use anymore b/c they will very soon be on the new stack, we can take out already. However, I undid all those examples removals that you questioned.

And yes, we will replace most of our examples with new stack equivalents, but also clean up this folder well (it's become pretty messy over time :( ).

**Review by kouroshHakha — APPROVED** [review]

**Pull request merged**